src/ORM/Association.php

```diff
@@ -400,6 +401,15 @@ public function getTarget()
                 $config = ['className' => $this->_className];
             }
             $this->_targetTable = $tableLocator->get($registryAlias, $config);
+
+            $className = $this->_getClassName($registryAlias, ['className' => $this->_className]);
+            if (!is_a($this->_targetTable, $className)) {
```

Comment:
OK so what if I request a `BarTable` and there's `FooTable` in a registry? Assuming `BarTable` extends `FooTable`. I'd still get a wrong table with `instanceof` if I understand correctly.